plugins/serializers/template/template.go

```diff
@@ -0,0 +1,100 @@
+package template
+
+import (
+	"fmt"
+	"strings"
+	"text/template"
+
+	"github.com/Masterminds/sprig/v3"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/serializers"
+)
+
+type Serializer struct {
+	Template      string          `toml:"template"`
+	BatchTemplate string          `toml:"batch_template"`
+	Log           telegraf.Logger `toml:"-"`
+
+	tmplMetric *template.Template
+	tmplBatch  *template.Template
+}
+
+func (s *Serializer) Init() error {
+	// Setting defaults
+	var err error
+
+	s.tmplMetric, err = template.New("template").Funcs(sprig.TxtFuncMap()).Parse(s.Template)
+	if err != nil {
+		return fmt.Errorf("creating template failed: %w", err)
+	}
+	if s.BatchTemplate == "" {
+		s.BatchTemplate = fmt.Sprintf("{{range .}}%s{{end}}", s.Template)
+	}
+	s.tmplBatch, err = template.New("batch template").Funcs(sprig.TxtFuncMap()).Parse(s.BatchTemplate)
+	if err != nil {
+		return fmt.Errorf("creating batch template failed: %w", err)
+	}
+	return nil
+}
+
+func (s *Serializer) Serialize(metric telegraf.Metric) ([]byte, error) {
+	m, ok := metric.(telegraf.TemplateMetric)
+	if !ok {
+		s.Log.Errorf("metric of type %T is not a template metric", metric)
+		return nil, nil
+	}
+	var b strings.Builder
+	if s.Template == "" && s.BatchTemplate != "" {
+		// If the user has only defined a batch template, fall back to batches of 1
+		metrics := []telegraf.TemplateMetric{m}
+
+		if err := s.tmplBatch.Execute(&b, &metrics); err != nil {
+			s.Log.Errorf("failed to execute batch template: %v", err)
+			return nil, nil
+		}
+	} else {
+		if err := s.tmplMetric.Execute(&b, &m); err != nil {
+			s.Log.Errorf("failed to execute template: %v", err)
+			return nil, nil
+		}
+	}
+
+	return []byte(b.String()), nil
```

Comment:
Wouldn't this be easier to understand
```suggestion
	// We got a metric-per-metric template so simply call it with the one
	// given metric.
	if s.Template != "" {
		if err := s.tmplMetric.Execute(&b, &m); err != nil {
			s.Log.Errorf("failed to execute template: %v", err)
			return nil, nil
		}
		return b.Bytes(), nil
	}
	
	// For batch templates we need to wrap the metric into a
	// slice so that the template can iterate over it.
	if s.BatchTemplate != "" {
		metrics := []telegraf.TemplateMetric{m}
		if err := s.tmplBatch.Execute(&b, &metrics); err != nil {
			s.Log.Errorf("failed to execute batch template: %v", err)
			return nil, nil
		}
		return b.Bytes(), nil
	}
	
	// No template at all
	return nil, nil
```

Reply:
👍